### PR TITLE
tracing for Acl,discovery,policy and relationship actors

### PR DESCRIPTION
**Pull request by swathisusi**

### CODE REVIEW

File: copper-server/src/main/java/com/pogeyan/cmis/actors/AclActor.java [hunk 1]
```diff
@@ -46,15 +51,24 @@ public String getName() {
...
+		ISpan parentSpan = (ISpan) baggage.get(BrowserConstants.PARENT_SPAN);
+		ISpan span = TracingApiServiceFactory.getApiService().startSpan(tracingId, parentSpan, "AclActor::applyAcl",
+				null);
+		Map<String, Object> attrMap = new HashMap<String, Object>();
 		if (!Helpers.checkingUserPremission(permission, "post")) {
+			attrMap.put("error", t.getUserName() + "is not authorized to applyAcl, TraceId:" + span.getTraceId());
+			TracingApiServiceFactory.getApiService().updateSpan(span, true,
+					t.getUserName() + " is not authorized to applyAcl", attrMap);
+			TracingApiServiceFactory.getApiService().endSpan(tracingId, span);
 			throw new CmisRuntimeException(t.getUserName() + " is not authorized to applyAcl.");
```
Comment (susmithavani30) [inline copper-server/src/main/java/com/pogeyan/cmis/actors/AclActor.java]: Throw traceId here also in exceptions

File: copper-server/src/main/java/com/pogeyan/cmis/actors/RelationshipActor.java [hunk 2]
```diff
@@ -75,15 +90,18 @@ private JSONObject getRelationships(QueryGetRequest request)
...
 				objectId, includeSubRelationshipTypes, relationshipDirection, typeId, renditionFilter,
-				includeAllowableActions, maxItems, skipCount, null, request.getUserObject());
-
+				includeAllowableActions, maxItems, skipCount, null, request.getUserObject(), tracingId, span);
 		if (relationships == null) {
+			attrMap.put("error", "Relationships are null!, TraceId:" + span.getTraceId());
+			TracingApiServiceFactory.getApiService().updateSpan(span, true,
+					request.getUserName() + " Relationships are null!", attrMap);
+			TracingApiServiceFactory.getApiService().endSpan(tracingId, span);
 			throw new CmisRuntimeException("Relationships are null!");
```
Comment (susmithavani30) [inline copper-server/src/main/java/com/pogeyan/cmis/actors/RelationshipActor.java]: same here

File: copper-server/src/main/java/com/pogeyan/cmis/actors/PolicyActor.java [hunk 3]
```diff
@@ -71,68 +89,96 @@ private JSONArray policies(QueryGetRequest request) throws CmisObjectNotFoundExc
...
 		String permission = request.getUserObject().getPermission();
+		Map<String, Object> attrMap = new HashMap<String, Object>();
 		if (!Helpers.checkingUserPremission(permission, "post")) {
-			throw new CmisRuntimeException(request.getUserName() + " is not authorized to applyAcl.");
+			attrMap.put("error",
+					request.getUserName() + "is not authorized to removePolicy, TraceId:" + span.getTraceId());
+			TracingApiServiceFactory.getApiService().updateSpan(span, true,
+					request.getUserName() + " is not authorized to removePolicy", attrMap);
+			TracingApiServiceFactory.getApiService().endSpan(tracingId, span);
+			throw new CmisRuntimeException(request.getUserName() + " is not authorized to removePolicy.");
```
Comment (susmithavani30) [inline copper-server/src/main/java/com/pogeyan/cmis/actors/PolicyActor.java]: same here

File: copper-server/src/main/java/com/pogeyan/cmis/actors/RelationshipActor.java [hunk 4]
```diff
@@ -46,16 +51,26 @@ public String getName() {
...
+				"RelationshipActor::getRelationships", null);
 		String permission = request.getUserObject().getPermission();
+		Map<String, Object> attrMap = new HashMap<String, Object>();
 		if (!Helpers.checkingUserPremission(permission, "get")) {
-			throw new CmisRuntimeException(request.getUserName() + " is not authorized to applyAcl.");
+			attrMap.put("error", request.getUserName() + "is not authorized, TraceId:" + span.getTraceId());
+			TracingApiServiceFactory.getApiService().updateSpan(span, true,
+					request.getUserName() + " is not authorized", attrMap);
+			TracingApiServiceFactory.getApiService().endSpan(tracingId, span);
+			throw new CmisRuntimeException(request.getUserName() + " is not authorized.");
```
Comment (susmithavani30) [inline copper-server/src/main/java/com/pogeyan/cmis/actors/RelationshipActor.java]: same here

File: copper-server-impl/src/main/java/com/pogeyan/cmis/impl/services/CmisDiscoveryService.java [hunk 5]
```diff
@@ -60,16 +60,23 @@
...
+			Map<String, Object> attrMap = new HashMap<String, Object>();
 			MDiscoveryServiceDAO discoveryObjectMorphiaDAO = DatabaseServiceFactory.getInstance(repositoryId)
 					.getObjectService(repositoryId, MDiscoveryServiceDAO.class);
 			MTypeManagerDAO typeManagerDAO = DatabaseServiceFactory.getInstance(repositoryId)
 					.getObjectService(repositoryId, MTypeManagerDAO.class);
 			int maxItemsInt = maxItems == null ? 10 : maxItems.intValue();
 			if (changeLogToken == null || changeLogToken.getValue() == null) {
-				throw new CmisInvalidArgumentException("change log token value should not be null!");
+				attrMap.put("error", "change log token value should not be null!, TraceId: " + span.getTraceId());
+				TracingApiServiceFactory.getApiService().updateSpan(span, true,
```
Comment (fahadsuhaib) [inline copper-server-impl/src/main/java/com/pogeyan/cmis/impl/services/CmisDiscoveryService.java]: We are setting the same text value in the attrMap and the function in updateSpan?

**Pull request closed without merging**